prov/gni/src/gnix_hashtable.c

```diff
@@ -126,22 +126,34 @@ static inline gnix_ht_key_t __gnix_hash_func(
 			ht->ht_attr.ht_hash_seed) % ht->ht_size;
 }
 
+static inline gnix_ht_entry_t *__gnix_ht_lookup_entry_collision(
+                struct dlist_entry *head,
+                gnix_ht_key_t key,
+                uint64_t *collision_count)
+{
+        gnix_ht_entry_t *ht_entry;
+
+        dlist_for_each(head, ht_entry, entry) {
+                READ_PREFETCH(head->next);
```

Comment:
I re-ran the numbers. The values are mostly unchanged, but tend to be more towards ~0.03 rather than 0.02usec. Slight improvement, as it was in the original branch.